cmake/instrument.cmake

```diff
@@ -0,0 +1,20 @@
+# https://llvm.org/docs/XRay.html
+
+option (ENABLE_XRAY "Enable LLVM XRay" OFF)
+
+if (NOT ENABLE_XRAY)
+    message (STATUS "Not using LLVM XRay")
+    return()
+endif()
+
+if (NOT (ARCH_AMD64 AND (OS_LINUX OR OS_FREEBSD)))
```

Comment:
I suppose FreeBSD is not very popular as an OS. Unless X-Ray-on-FreeBSD is tested and verified to work, I suggest to support X-Ray only on Linux.